File: app/src/main/java/br/com/vicentec12/mygames/data/source/game/GameDao.java
package br.com.vicentec12.mygames.data.source.game;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import br.com.vicentec12.mygames.data.model.Game;

@Dao
public interface GameDao {

    @Query("SELECT * FROM game WHERE id_console = :idConsole ORDER BY name")
    List<Game> list(long idConsole);

    @Query("SELECT * FROM game WHERE id_console = :idConsole ORDER BY year")
    List<Game> listByYear(long idConsole);

    @Query("SELECT * FROM game WHERE id = :id")
    Game get(int id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insert(Game game);

    @Update
    int update(Game game);

    @Delete
    int delete(Game game);

    @Delete
    int delete(List<Game> games);

    @Query("DELETE FROM game")
    void deleteAll();

}

File: app/src/main/java/br/com/vicentec12/mygames/data/source/game/GameDao.kt
package br.com.vicentec12.mygames.data.source.game

import androidx.room.*
import br.com.vicentec12.mygames.data.model.Game

@Dao
interface GameDao {

    @Query("SELECT * FROM game WHERE console_id = :mIdConsole ORDER BY name")
    suspend fun list(mIdConsole: Long): List<Game?>?

    @Query("SELECT * FROM game WHERE console_id = :mIdConsole ORDER BY year")
    suspend fun listByYear(mIdConsole: Long): List<Game?>?

    @Query("SELECT * FROM game WHERE id = :mId")
    suspend fun get(mId: Long): Game?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(mGame: Game?): Long

    @Update
    suspend fun update(mGame: Game?): Int

    @Delete
    suspend fun delete(mGames: List<Game?>?): Int

    @Query("DELETE FROM game")
    suspend fun deleteAll()

}